client/packages/invoices/src/Returns/InboundDetailView/DetailView.tsx

```diff
@@ -88,10 +77,18 @@ export const InboundReturnDetailView: FC = () => {
               returnId={data.id}
               initialItemId={itemId}
               modalMode={mode}
-              loadNextItem={nextItemId ? () => onOpen(nextItemId) : undefined}
+              loadNextItem={() => {
+                if (nextItemId) onOpen(nextItemId);
+                else {
+                  // Closing and re-opening forces the modal to launch with the
+                  // item selector in focus
+                  onClose();
+                  setTimeout(() => onOpen(), 50);
+                }
```

Comment:
I don't love this, but I couldn't find any other way to get the new item modal to launch with the item selector opened/in focus (without adding a bunch of complexity to the ItemSelector component). I don't fully understand how they've done it in the Shipments, but this approach actually gives a less jarring visual jump than the Shipment one anyway, so 🤷‍♂️

Reply:
Interesting... the other modals use a method that returns `true` which is captured by `useDialog` to do the little slide animation, but I guess you also get that here by forcing another `onOpen`. This does feel a bit hacky, but also less magic, which I like. I'd be curious to see what other people in the team think!

Reply:
Can you point to where this is done -- I'm not seeing it, sorry.

Reply:
https://github.com/msupply-foundation/open-msupply/blob/5c125e06c4036edc3ad04f5a958e01a6fed907a7/client/packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/InboundLineEdit.tsx#L153-L165

https://github.com/msupply-foundation/open-msupply/blob/5c125e06c4036edc3ad04f5a958e01a6fed907a7/client/packages/common/src/hooks/useDialog/useDialog.tsx#L187